homeassistant/components/plex/server.py

```diff
@@ -669,6 +669,120 @@ def lookup_media(self, media_type, **kwargs):
             _LOGGER.error("%s not found in %s", failed_item, library_name)
             return None
 
+    def continue_media(self, media, library_name, continuation):
+        """Find the episode object from which to resume watching.
+
+        The following continuation modes are supported:
+        1. deck
+        2. unfinished (episodes that were watched partway through)
+        3. unwatched
+        4. watched
+
+        Each mode above can function as a fallback in case the prior mode
+        fails to find suitable media. Fallbacks are delimited by a "|". If
+        all modes fail to find media, the media will start from the beginning.
+
+        Modes may also be parameterized to get the 'first' or 'last' element
+        among the results (e.g. unwatched_first, unfinished_last). Parameters are
+        optional and are delimited by an "_". Default parameter is 'first'.
+
+        Notes:
+            - Deck mode supports shows, seasons
+            - Unfinished mode supports shows, seasons
+            - Unwatched mode supports shows, seasons
+            - Watched mode supports shows, seasons
+
+        Args:
+            media (plexapi.video.Show or plexapi.video.Season): media to be continued
+            library_name (str): library section used for 'deck' continuation mode
+            continuation (str): structured continuation query as described in this method
+
+        Returns:
+            plexapi.video.Show or plexapi.video.Season object
+
+            Return object will always be either the original media argument, or a more suitable
+            continuation media.
+
+        Examples:
+            Let's assume the following complex continuation mechanism is desired:
+
+                Resume media from deck, and if it's missing then
+                resume last unfinished episode, and if it's missing then
+                resume first unwatched episode, and if it's missing then
+                start show from the beginning
+
+            Resultant continuation query:
+
+                "_continue": "deck|unfinished_last|unwatched_first"
+
+        """
+        if type(media) not in (plexapi.video.Show, plexapi.video.Season):
+            return media
+
+        if not continuation or not media:
+            return media
+
+        for query in continuation.lower().split("|"):
+            # Query format: <mode>[_<parameter>]
+            args = query.split("_")
+            mode = args[0]
+            index = 0
+
+            # Validate mode parameters
+            if len(args) == 2 and args[1] in ("first", "last"):
+                index = 0 if args[1] == "first" else -1
+            elif len(args) != 1:
+                _LOGGER.error("Unrecognized continuation mode args: '%s'", query)
+                continue
+
+            episodes = []
+            if mode == "deck":
```

Comment:
Perhaps the possible `mode` values could be made into a lookup dict or `Enum`?

Reply:
I'm deeply sorry, but I wasn't entirely sure how to interpret this suggestion (I understand literally, but not entirely sure how it's intended to be applied). May I ask to take a look at the updated code shortly to see if dict/enum is needed? and if so, how would it be applied? Just translating string -> enum and then if/then check based on enum value instead of string comparison?

Very sorry if I'm missing something obvious.

Reply:
I forget how it looked before, but the current way seems just fine.